source/common/http/conn_manager_impl.cc

```diff
@@ -662,9 +662,6 @@ ConnectionManagerImpl::ActiveStream::ActiveStream(ConnectionManagerImpl& connect
     max_stream_duration_timer_->enableTimer(connection_manager_.config_.maxStreamDuration().value(),
                                             this);
   }
-
-  filter_manager_.streamInfo().setRequestedServerName(
```

Comment:
I thought this was being done per http request. I'm not sure this one can be removed. But OTOH, all the tests passed. @mattklein123 is this really not needed, or are we just lacking test coverage somewhere?

Reply:
Yes I think this is likely a bug in the same class of issues I fixed a while back with addresses. Optimally, we would do something similar here so we don't mess this up.

Reply:
I'm 99% sure that deleting this line will break things, but apparently we don't have any test coverage of this. I think this line needs to be reverted.

Reply:
Yes, I dig into it. Then I realized this isn't the same streamInfo with the L4 layer one. This streamInfo is an instance inside ActiveStream, although part of its info is coming from the L4 layer(include this server name). So I shouldn't remove it.

I also added unitest to cover this case https://github.com/envoyproxy/envoy/pull/16797